### DISCUSSION TITLE
Feedback!

### DISCUSSION THREAD

**Discussion by PatrickAlphaC** [Cyfrin/security-and-auditing-full-course-s23#175] [General]
In this thread, please leave feedback on your favorite parts, and least favorite parts of the course!

Notes:
1. Always have a feedback discussion
2. Add signatures section
3. Exploit lessons are Travis's favorite (sc-exploits-minimized)
4. Have more challenges
5. PIN DEPENDENCIES

**Comment by PatrickAlphaC** [comment]
https://github.com/Cyfrin/security-and-auditing-full-course-s23/issues/179

Notes from @Equious  ^

- The high level coverage of signatures is fine for the purposes of the content, but it leaves quite a few unanswered questions such as the actual definitions and derivations of the v, r and s values. Expanding with content focused on signatures would be good!
- MOAR Remix/Vulnerability examples. Seeing the bug in action is awesome.

**Comment by al0x23** [comment]
I don't know if it's the right place, but in lesson 26 of the TSwap - Security course the description doesn't seem very accurate 

"One Last Huzzah
Patrick explains how fuzz testing **and formal verification methods can be used together** to detect bugs in Solidity smart contracts."

I didn’t catch any mention of formal verification in the video, but maybe I just didn’t understand at all what I was looking at

**Reply by EngrPips** [reply]
You can make a PR on the official GitHub repo of Cyfrin Updraft to correct any inconsistencies.

**Reply by PatrickAlphaC** [reply]
Ah yes, we should fix that.

**Comment by Kaydo1-tevikay** [comment]
how do i get to see a bug in a project (smart contract) i reviewed four first flight now buh i only read others bug report and the smart contract https://updraft.cyfrin.io/courses/security is frustracting to me now. how do i start work

**Reply by EngrPips** [reply]
with practice you know better and you see better, there is no magic or trick to catching bugs.

**Comment by blastonyz** [comment]
Many thanks to the Cyfrin team for all the material. Speaking of favorite parts, the explanation of fuzzing was undoubtedly epic; I think PuppyRaffle marks a real turning point—using analysis tools, solidifying the mindset, and walking through the project. A big shout-out to one of my new heroes, Patrick the Wizard, @PatrickAlphaC . 
It’s not just about the content; motivation changes lives, and we can all always use a little bit of it. 
Always grateful for this

**Reply by EngrPips** [reply]
Awesome, we are glad that you appreciate the enormous work that went into creating the course content.
